src/Instrumentation/Session/src/PhpSessionInstrumentation.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\Instrumentation\PhpSession;
+
+use OpenTelemetry\API\Instrumentation\CachedInstrumentation;
+use OpenTelemetry\API\Trace\Span;
+use OpenTelemetry\API\Trace\SpanBuilderInterface;
+use OpenTelemetry\API\Trace\SpanKind;
+use OpenTelemetry\API\Trace\StatusCode;
+use OpenTelemetry\Context\Context;
+use function OpenTelemetry\Instrumentation\hook;
+use OpenTelemetry\SemConv\TraceAttributes;
+use OpenTelemetry\SemConv\Version;
+use Throwable;
+
+/** @psalm-suppress UnusedClass */
+class PhpSessionInstrumentation
+{
+    public const NAME = 'php-session';
+
+    public static function register(): void
+    {
+        $instrumentation = new CachedInstrumentation(
+            'io.opentelemetry.contrib.php.session',
+            null,
+            Version::VERSION_1_36_0->url(),
+        );
+
+        self::_hook($instrumentation, null, 'session_start', 'session.start');
+        self::_hook($instrumentation, null, 'session_destroy', 'session.destroy');
+        self::_hook($instrumentation, null, 'session_write_close', 'session.write_close');
+        self::_hook($instrumentation, null, 'session_unset', 'session.unset');
+        self::_hook($instrumentation, null, 'session_abort', 'session.abort');
+    }
+
+    /**
+     * Simple generic hook function which starts and ends a minimal span
+     * @psalm-suppress UnusedFunctionCall
+     */
+    private static function _hook(CachedInstrumentation $instrumentation, ?string $class, string $function, string $name): void
+    {
+        hook(
+            class: $class,
+            function: $function,
+            /** @psalm-suppress UnusedClosureParam */
+            pre: static function ($object, ?array $params, ?string $class, string $function, ?string $filename, ?int $lineno) use ($instrumentation, $name) {
+                /** @psalm-suppress ArgumentTypeCoercion */
+                $builder = self::makeBuilder($instrumentation, $name, $function);
+                self::addParams($builder, $function, $params);
+                $parent = Context::getCurrent();
+                $span = $builder->startSpan();
+                Context::storage()->attach($span->storeInContext($parent));
+            },
+            post: static function ($object, ?array $params, mixed $return, ?Throwable $exception) use ($instrumentation, $name, $function) {
+                $builder = self::makeBuilder($instrumentation, $name, $function);
```

Comment:
I'm a bit confused about this. You create a builder (as if a new span was going to be started), but then don't do anything with it. Normally I'd expect a `pre` hook to start a span, and the `post` hook to end the active span, but not create any new spans...